pkg/cmd/backupdriver/cli/install/install.go

```diff
@@ -285,3 +225,34 @@ func (o *InstallOptions) Complete(args []string, f client.Factory) error {
 
 	return nil
 }
+
+func (o *InstallOptions) CheckClusterFlavorForBackupDriver() error {
+	clusterFlavor, err := utils.GetClusterFlavor(nil)
+	if err != nil {
+		return errors.Wrap(err, "Failed to get cluster flavor for backup-driver")
+	}
+
+	// Assign master node affinity and host network to Supervisor deployment
+	if clusterFlavor == constants.Supervisor {
+		fmt.Printf("Supervisor Cluster. Assign master node affinity and enable host network.")
+		o.MasterAffinity = true
+		o.HostNetwork = true
+	}
+
+	return nil
+}
+
+func (o *InstallOptions) CheckFeatureFlagsForBackupDriver(kubeClient kubernetes.Interface) error {
+	featureFlags, err := cmd.GetVeleroFeatureFlags(kubeClient, o.Namespace)
+	if err != nil {
+		fmt.Printf("Failed to decipher velero feature flags: %v, assuming none.\n", err)
+	}
+
+	// Create Config Map with the feature states.
+	if err := cmd.CreateFeatureStateConfigMap(kubeClient, featureFlags, o.Namespace); err != nil {
+		fmt.Printf("Failed to create feature state config map from velero feature flags: %v.\n", err)
+		return err
+	}
```

Comment:
Lets stick to one format, most of the code in our plugin explicitly checks for `err!=nil` instead of this one liner.

Reply:
I thought it is better to use one-line format here and in similar cases. Because, `err` in this case will be only scoped within the `if` block, which prevent any side-effect of the `err` on the following context.

Reply:
I don't see it specifically giving any benefits in this case, `CreateFeatureStateConfigMap` returning error would overwrite the err, which is fine. The idea is to have uniformity all across the code, and in one of my reviews, Dave suggested we stick to checking err separately.

Reply:
It may not show remarkable benefit in this case. But, it would be better overall.

1. make code simpler and more readable.
2. constrain the err variable within the scope of if block, which prevent any side-effect.
3. less error-prone in coding.

For example, if we check error separately, it is much more easier for us to forget to check the error afterwards. We saw a lot of such cases in the code base of velero-plugin-for-vsphere. The one-line format would be effective to help avoid this issue.

So, I would recommend to use the one-line format and have the uniformity by trying to change existing code to follow this fashion.

Reply:
astrolabe uses the other format. Anyways, I seem to get the benefits now, we can use the one-liner format for functions that return only error, for others we could use the separate error check.